apps/choleskey/cholesky_tiled.cpp

```diff
@@ -0,0 +1,192 @@
+#include <bits/stdc++.h>
+#include <cblas.h>
+#include <lapacke.h>
+#include <omp.h>
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <experimental/linalg>
+#include <iostream>
+#include <utility>
+#include <vector>
+#include "help.hpp"
+
+using data_type = double;
+
+//#include "time.hpp"
+#define SWITCH_CHAR '-'
+
+double* dpo_generate(size_t side_size) {
+    unsigned int seed = side_size;
+#ifdef _WIN32
+    srand(seed);
+#endif
+    // M is a (very) pseudo-random symmetric matrix
+    double* M_matrix = new double[side_size * side_size];
+    for (size_t row = 0; row < side_size; ++row) {
+        for (size_t col = row; col < side_size; ++col) {
+            M_matrix[col * side_size + row] = M_matrix[row * side_size + col] = (double)
+#ifdef _WIN32
+                                                                                    rand()
+#else
+                                                                                    rand_r(&seed)
+#endif
```

Comment:
are there any reason putting _WIN32 here? I think our code will not run on windows machine.

Reply:
yes, put the unmodified version here. will replace it.